mlir/lib/Dialect/PDL/IR/Builtins.cpp

```diff
@@ -52,5 +121,6 @@ void registerBuiltins(PDLPatternModule &pdlPattern) {
                                      createArrayAttr);
   pdlPattern.registerRewriteFunction("__builtin_addElemToArrayAttr",
                                      addElemToArrayAttr);
+  pdlPattern.registerConstraintFunctionWithResults("__builtin_add", add);
```

Comment:
Can we also have it in the rewrite section and have a test for it?
`registerRewriteFunction`.